### PR TITLE
feat: staking endpoint support

### LINKED ISSUE #786: Add staking support
Since Bee version 1.9 there is support for staking with the `/stake` endpoint. It would be good if bee-js also supported that.

### PR DESCRIPTION
**Pull request by AuHau**
Closes #786

### CODE REVIEW

File: src/modules/debug/stake.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+export async function stake(ky: Ky, amount: NumberString, options?: TransactionOptions): Promise<void> {
+  const headers: Record<string, string> = {}
+
+  if (options?.gasPrice) {
+    headers['gas-price'] = options.gasPrice.toString()
+  }
+
+  if (options?.gasLimit) {
+    headers['gas-limit'] = options.gasLimit.toString()
+  }
```
Comment (Cafe137) [inline src/modules/debug/stake.ts:41]: Is this not affected by the EIP-1559 stuff?

**Review by Cafe137 — APPROVED** [review]

File: src/modules/debug/stake.ts (hunk 1 above)
Comment (AuHau) [reply src/modules/debug/stake.ts:41]: No idea, but this is supported on API, so I have added the support here as well.

File: test/integration/bee-debug-class.spec.ts [hunk 2]
```diff
@@ -143,6 +143,26 @@ describe('Bee Debug class', () => {
...
+    it(
+      'should deposit stake',
+      async () => {
+        const originalStake = BigInt(await beeDebug.getStake())
+
+        await beeDebug.depositStake('100000000000000000')
+
+        const increasedStake = BigInt(await beeDebug.getStake())
+
+        expect(increasedStake - originalStake === BigInt(10e16)).toBeTruthy()
```
[the remark this answers is not in the document]
Comment (AuHau) [reply test/integration/bee-debug-class.spec.ts:160]: I am not sure how Jest plays with BigInt comparison, so I choose this path, but I will try the clean path instead.

[the remark this answers is not in the document]
Comment (AuHau) [reply test/integration/bee-debug-class.spec.ts]: It worked ;-)

File: src/modules/debug/stake.ts (hunk 1 above)
Comment (agazso) [reply src/modules/debug/stake.ts:41]: EIP-1559 will arrive in Bee 1.10

**Comment by bee-worker** [comment]
Release [v5.1.0](https://github.com/ethersphere/bee-js/releases/tag/v5.1.0) addresses this.

**Pull request merged**